first setyp sync documentation (#281)
FirelyTeam/firely-docs-firely-server: features_and_tools/pubsub.rst

* ``BrokerType`` - The message broker that you want to use, either AzureServiceBus or RabbitMQ
* ``ResourceChangeNotifications`` - This section can be enabled to let Firely Server send out information on changes to the message broker, so that other services can sync with Firely Server. Note that this is only available for Firely Server instances that use SQL server (2016 and newer) as a repository database. As of yet it cannot work in combination with MongoDB or SQLite.
* ``SendLightEvents`` - Either set to true or false, this will enable Firely Server to send out messages with information on changes, these messages will not contain the complete resource.
* ``SendFullEvents`` - Either set to true or false, this will enable Firely Server to send out messages with information on changes, these messages contain the complete resource.
* ``PollingIntervalSeconds`` - By default set to five seconds, Firely Server will check the database every five seconds for changes. 
* ``MaxPublishBatchSize`` - The maximum amount of resources that can be send to the message broker at once.

Message types and formats
-------------------------

> PubSub expects standard message formats for a set of different messages that can be communicated to RabbitMQ. When other services want to receive updates on changes they can expect messages also to take on a set format. 

Reviewer: I was thinking about rephrasing it to something like...
```suggestion
To establish communication between Firely Server and other applications, the parties must share the same contract. This section describes the contract.
```

PubSub uses MassTransit to envelope these messages using specific headers, for documentation on these headers we refer to the `MassTransit documentation page <https://masstransit.io/documentation/concepts/messages#message-headers>`_.
Below we give an overview of the different message types that can be communicated with PubSub.

> ExecuteStorePlan

Reviewer: ```suggestion
ExecuteStorePlanCommand
```

^^^^^^^^^^^^^^^^

> This message can be send to the message broker by your client to let Firely Server execute a batch of instructions to create, update, upsert, or delete resources that should be processed as a transaction, so either

Reviewer: ```suggestion
This command can be sent to the message broker by your client to let Firely Server execute a batch of instructions to create, update, upsert, or delete resources that should be processed as a transaction, so either
```

all of the instructions are performed, or none. The instructions are CRUD-type operations that operate on a store of resources, each with its own id. These ids are unique per type of resource.
Note that this message should only contain one operation per resource (so per resource type + id) as the operations in the message are supposed to bring each resource involved to its desired final state, rather than reflect a set of operations that would present a history of operations on a resource.

.. code-block::

      "message": {
        "instructions": [
            {
            "itemId": "Patient/testid",
            "resource": "{\"resourceType\":\"Patient\",\"id\":\"testid\",\"meta\":{\"versionId\":\"versionId=test\",\"lastUpdated\":\"2023-10-09T12:00:22.8990506+02:00\"},\"name\":[{\"family\":\"id=test\"}]}",
            "resourceType": "",
            "resourceId": "",
            "currentVersion": "",
            "operation": create
        }
        ]
    }

> The ``ExecuteStorePlan`` message typically contains an array of instructions, where each instruction can contain the following fields:

Reviewer: ```suggestion
The ``ExecuteStorePlanCommand`` message typically contains an array of instructions, where each instruction can contain the following fields:
```

  "messageId": "<automatically assigned id>",
  "requestId": "<automatically assigned id>",
  "correlationId": null,
  "conversationId": "<automatically assigned id>",
  "initiatorId": null,
  "sourceAddress": "<source address of the message, this may be a temporary address for messages sent from IBus>",
  "destinationAddress": "<where the message was sent, such as an exchange in RabbitMQ>",
  "responseAddress": "<where responses to the request should be sent>",
  "faultAddress": null,
  "messageType": [
    "urn:message:Firely.Server.Contracts.Messages.V1:ExecuteStorePlanCommand"
  ],
  "message": {
    "instructions": [
      {
        "itemId": "Patient/1",
        "resource": "{\"resourceType\":\"Patient\",\"id\":\"1\",\"meta\":{\"lastUpdated\":\"2023-10-15T12:35:54.5070437+02:00\"},\"name\":[{\"family\":\"test\"}]}",
        "resourceType": null,
        "resourceId": null,
        "currentVersion": null,
        "operation": "create"
      }
    ]
  },

  "expirationTime": "2023-10-15T10:36:55.0495563Z",
  "sentTime": "2023-10-15T10:35:54.8989292Z",
  "headers": {
    "MT-Request-AcceptType": [
      "urn:message:Firely.Server.Contracts.Messages.V1:ExecuteStorePlanResponse"

>     ]

Reviewer 2: ```suggestion
    ],
```
